junit-jupiter-api/src/main/java/org/junit/jupiter/api/AssertionUtils.java

```diff
@@ -93,7 +94,15 @@ private static String formatClassAndValue(Object value, String valueString) {
 	}
 
 	private static String toString(Object obj) {
-		return (obj instanceof Class ? getCanonicalName((Class<?>) obj) : String.valueOf(obj));
+		if (obj instanceof Class) {
+			return getCanonicalName((Class<?>) obj);
+		}
+		else if (obj instanceof Object[]) {
+			return Arrays.toString((Object[]) obj);
+		}
+		else {
+			return String.valueOf(obj);
+		}
```

Comment:
> Can you please move this to a separate PR outlining the rationale for the change?

See https://github.com/junit-team/junit5/pull/961/files#r134096347 for the rationale. :)